torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -86,24 +85,24 @@ def __init__(self,
         else:
             self.datapipe = datapipe.unbatch(unbatch_level=unbatch_level)
         self.buffer_size = buffer_size
-        self._buffer = []
 
-    def buffer_replace(self, x):
-        idx = random.randint(0, self.buffer_size - 1)
-        val = self._buffer[idx]
-        self._buffer[idx] = x
+    @staticmethod
+    def buffer_replace(buffer, x):
```

Comment:
Nit: In this implementation, `x` cannot be randomly selected and returned.
Another possible implementation is that you draw a number from `[0, len(buffer]` and return `x` if `idx == len(buffer)`. In that case, there can be some shuffling even when `buffer_size = 1` (but no shuffling at all if `buffer_size = 0 `.

Reply:
I think it depends on how we define the buffer.
I will try out TF's buffer shuffle with buffer size 1.